### PR TITLE
Clean up lidar and lane detect

### PR DESCRIPTION
**Pull request by phucla0122**
- There are not too many things that can be cleaned up for these files since they are pretty well made.
- I only changed a couple of things in the lane_dectect.py file by moving some constants and making them global variables
- I also separate the test files and put them into the test folder
- The testing itself needs some work on it since the test right now just writes the calculated value in the metrics.txt files. The next step is to make it automate, which could be challenging since the numbers are magic numbers.
- For lidar, not much change besides moving the queue_size variable in lidar_object_detect.py to become a global variable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ executables for MacOS and only some for Linux.
     > 
     > There is a chance of getting this message in the ROS console: 31mRLException: [full_system.launch] is neither a launch file in package [central_control] nor is [central_control] a launch file name.
       To fix this, make sure the cloned repository's filepath does not contain any whitespace, then restart this step. The build and devel folders may need to be deleted first.
+     To fix this, search the code for "airsim.CarClient()" and replace it with "airsim.CarClient()" .
```
Comment (nicolastuttle) [inline README.md]: Was this copied by accident?

File: src/central_control/scripts/central_control.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ class CentralControl:
     def __init__(self):
         host_ip = rospy.get_param('/host_ip')
 
-        self.client = airsim.CarClient(ip=host_ip)
+        self.client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/central_control/scripts/central_control.py]: You can revert all these `ip=host_ip` changes

File: src/lane_keep_assist/run_metrics.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from lane_test import lane_detect_test
+from testing.PythonTests.lane_keep_assist import lane_detect_test
```
Comment (nicolastuttle) [inline src/lane_keep_assist/run_metrics.py:1]: Testing will go on a different branch as discussed

File: src/lidar_object_recognition/scripts/cluster_detection.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def __init__(self):
 
         host_ip = rospy.get_param('/host_ip')
 
-        self.client = airsim.CarClient(ip=host_ip)
+        self.client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/lidar_object_recognition/scripts/cluster_detection.py]: host_ip

File: src/mapping_navigation/scripts/navigation.py [hunk 5]
```diff
@@ -108,7 +108,7 @@ def get_steering_angle(self) -> float:
 # Can use either of the path methods by uncommenting one or the other if you want to test with the old path
 if __name__ == "__main__":
 
-    map = load_from_file(os.path.abspath(os.path.dirname(__file__)) + "/new_coords.pickle")
+    map = load_from_file(os.path.abspath(os.path.dirname(__file__)) + "/test_new_coords.pickle")
```
Comment (nicolastuttle) [inline src/mapping_navigation/scripts/navigation.py]: Revert to original value

File: src/sensors/scripts/car_image_raw.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ def airpub():
 
     # connect to the AirSim simulator 
     host_ip = rospy.get_param('/host_ip')
-    client = airsim.CarClient(ip=host_ip)
+    client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/sensors/scripts/car_image_raw.py]: host_ip

File: src/sensors/scripts/car_pose.py [hunk 7]
```diff
@@ -11,7 +11,7 @@ def airpub():
 
     # connect to the AirSim simulator
     host_ip = rospy.get_param('/host_ip')
-    client = airsim.CarClient(ip=host_ip)
+    client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/sensors/scripts/car_pose.py]: host_ip

File: src/sensors/scripts/lidar.py [hunk 8]
```diff
@@ -20,7 +20,7 @@ def send_lidar_data(self):
 
         host_ip = rospy.get_param('/host_ip')
 
-        client = airsim.CarClient(ip=host_ip)
+        client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/sensors/scripts/lidar.py]: host_ip

File: src/sensors/scripts/segmented_image.py [hunk 9]
```diff
@@ -17,7 +17,7 @@ def segmented_image():
 
     # connect to the AirSim simulator
     host_ip = rospy.get_param('/host_ip')
-    client = airsim.CarClient(ip=host_ip)
+    client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/sensors/scripts/segmented_image.py]: host_ip

File: src/sensors/scripts/speedometer.py [hunk 10]
```diff
@@ -12,7 +12,7 @@ def speedometer():
 
     # connect to the AirSim simulator 
     host_ip = rospy.get_param('/host_ip')
-    client = airsim.CarClient(ip=host_ip)
+    client = airsim.CarClient()
```
Comment (nicolastuttle) [inline src/sensors/scripts/speedometer.py]: host_ip

**Review by nicolastuttle — CHANGES_REQUESTED** [review]
Good changes! Just the workarounds to undo and add `.idea` to `.gitignore` and I think we'll be ready to merge.

**Comment by nicolastuttle** [comment]
Looks like your find and replace was a bit too aggressive and you changed other files, can you undo those?

**Review by nicolastuttle — APPROVED** [review]

**Review by robellgabriel — APPROVED** [review]

**Pull request merged**